rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,212 @@
+from rpmlint.permissions import PermissionsParser, VariablesHandler
+
+
+class SUIDPermissionsCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.perms = {}
+
+        self.var_handler = VariablesHandler('/usr/share/permissions/variables.conf')
+
+        for fname in self._paths_to('permissions', 'permissions.secure'):
+            if not os.path.exists(fname):
+                continue
+
+            self._parse_profile(fname)
+
+    def _parse_profile(self, path):
+        parser = PermissionsParser(self.var_handler, path)
+        self.perms.update(parser.getEntries())
+
+    def _is_static_entry(self, entry):
+        # entries coming from the fixed permissions profile are considered
+        # static
+        return entry.profile.endswith('/permissions')
+
+    @staticmethod
+    def _paths_to(*file_names):
+        # we used to store the permissions data in /etc even though they aren't configuration files
+        # the whitelisting should check both paths (old /etc and new /usr/share) until all
+        # distributions using the old one (SLE15) are retired
+        for name in file_names:
+            # return the new path first.
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        need_set_permissions = False
+        found_suseconfig = False
+
+        for f, pkgfile in pkg.files.items():
+
+            if pkgfile.filecaps:
+                self.output.add_info(
+                    'E',
+                    pkg,
+                    'permissions-fscaps',
+                    "%(fname)s has fscaps '%(caps)s'"
+                    % {'fname': f, 'caps': pkgfile.filecaps},
+                )
+
+            mode = pkgfile.mode
+            owner = pkgfile.user + ':' + pkgfile.group
+
+            need_verifyscript = False
+            if f in self.perms or (stat.S_ISDIR(mode) and f + '/' in self.perms):
+                if stat.S_ISLNK(mode):
+                    self.output.add_info('W', pkg, 'permissions-symlink', f)
+                    continue
+
+                need_verifyscript = True
+
+                m = 0
+                o = 'invalid'
+                if stat.S_ISDIR(mode):
+                    if f in self.perms:
+                        self.output.add_info(
+                            'W', pkg, 'permissions-dir-without-slash', f
+                        )
+                    else:
+                        f += '/'
+
+                entry = self.perms[f]
+
+                if stat.S_ISREG(mode) and mode & (
+                    stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
+                ):
+                    # pie binaries have 'shared object' here
```

Comment:
I think we can drop this PIE binary check here. rpmlint2 has a standard check for that in BinariesCheck.py `_check_non_pie()`.